### PR TITLE
Human-readable tostring(mpack.NIL)

### PR DESCRIPTION
**Pull request by justinmk**
In lua, mpack.NIL is different than Lua nil. But that is confusing for users who might expect msgpack NIL to deserialize to Lua nil. Human-readable string representation helps avoid that confusion.

### CODE REVIEW

File: binding/lua/Makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ USE_SYSTEM_LUA ?= no
 LUA_VERSION_MAJ_MIN ?= 5.1
 LUA_VERSION ?= $(LUA_VERSION_MAJ_MIN).5
 LUA_VERSION_NOPATCH = $(shell echo -n $(LUA_VERSION) | sed 's!\([0-9]\.[0-9]\).[0-9]!\1!')
-LUA_URL ?= https://github.com/lua/lua/releases/download/$(LUA_VERSION)/lua-$(LUA_VERSION).tar.gz
+LUA_URL ?= https://www.lua.org/ftp/lua-$(LUA_VERSION).tar.gz
```
Comment (justinmk) [inline binding/lua/Makefile]: The old URL seems broken.

Comment (justinmk) [inline binding/lua/Makefile]: We made a similar change in neovim core: https://github.com/neovim/neovim/pull/5715

**Comment by justinmk** [comment]
Rebased

**Comment by tarruda** [comment]
@justinmk can you give me permission to edit this branch? I'd like to rebase and make a couple of changes before merging

**Comment by justinmk** [comment]
@tarruda Done

**Comment by tarruda** [comment]
merged, thanks

I will publish a new version for lua binding after #21 is merged

**Comment by justinmk** [comment]
Thanks!

**Pull request merged**